lib/ansible/modules/cloud/amazon/elb_application_lb.py

```diff
@@ -0,0 +1,778 @@
+            if new_load_balancer:
+                connection.delete_load_balancer(LoadBalancerArn=elb['LoadBalancerArn'])
+            module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+
+    # Now, if required, set ELB listeners. Use try statement here so we can remove the ELB if this stage fails
+    try:
+        listener_changed = create_or_update_elb_listeners(connection, module, elb)
+        if listener_changed:
+            changed = True
+    except ClientError as e:
+        # Something went wrong setting listeners. If this ELB was created during this task, delete it to leave a consistent state
+        if new_load_balancer:
+            connection.delete_load_balancer(LoadBalancerArn=elb['LoadBalancerArn'])
+        module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+
+    # Get the ELB again
+    elb = get_elb(connection, module)
+
+    # Get the ELB attributes again
+    elb.update(get_elb_attributes(connection, module, elb['LoadBalancerArn']))
+
+    # Convert to snake_case
+    snaked_elb = camel_dict_to_snake_dict(elb)
+
+    # Get the tags of the ELB
+    elb_tags = connection.describe_tags(ResourceArns=[elb['LoadBalancerArn']])['TagDescriptions'][0]['Tags']
+    snaked_elb['tags'] = boto3_tag_list_to_ansible_dict(elb_tags)
+
+    module.exit_json(changed=changed, **snaked_elb)
+
+
+def delete_elb(connection, module):
+
+    changed = False
+    elb = get_elb(connection, module)
+
+    if elb:
+        try:
+            connection.delete_load_balancer(LoadBalancerArn=elb['LoadBalancerArn'])
+            changed = True
+        except ClientError as e:
+            module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+        except NoCredentialsError as e:
+            module.fail_json(msg="AWS authentication problem. " + e.message, exception=traceback.format_exc())
+
+    module.exit_json(changed=changed)
+
+
+def main():
+
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            access_logs_enabled=dict(required=False, type='bool'),
+            access_logs_s3_bucket=dict(required=False, type='str'),
+            access_logs_s3_prefix=dict(required=False, type='str'),
+            deletion_protection=dict(required=False, default=False, type='bool'),
+            idle_timeout=dict(required=False, type='int'),
+            listeners=dict(required=False, type='list'),
+            name=dict(required=True, type='str'),
+            purge_tags=dict(required=False, default=True, type='bool'),
+            subnets=dict(required=False, type='list'),
+            security_groups=dict(required=False, type='list'),
+            scheme=dict(required=False, default='internet-facing', choices=['internet-facing', 'internal']),
+            state=dict(required=True, choices=['present', 'absent'], type='str'),
```

Comment:
Can `state` default to `present`? I usually expect this, I'm not sure if the flaw is with me or not :)